### PR TITLE
user-lockout documentation changes

### PR DESCRIPTION
**Pull request by akshya96**
JIRA: https://hashicorp.atlassian.net/browse/VAULT-12195
Related Epic: https://hashicorp.atlassian.net/browse/VAULT-8303

Includes the following changes:
User Lockout Concepts page
Configuration to include user-lockout change
User Lockouts configuration page
User lockout api page
Change user lockout config using auth tune (command and api) 
Telemetry

### CODE REVIEW

File: website/content/docs/concepts/user-lockout.mdx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+The precedence for user lockout disable is as follows: 
+
+Disable using environment variable VAULT_DISABLE_USER_LOCKOUT >> 
+Configuration for an auth mount using tune >> Configuration for an auth method in config file >> 
+Configuration for "all" auth methods in config file >> Default values.
+
+## Configuration
+
+User lockout parameters be configured using config file for "all" auth methods or a specific auth method (userpass, ldap or approle).
```
Comment (mladlow) [inline website/content/docs/concepts/user-lockout.mdx:29]:
```suggestion
User lockout parameters can be configured using config file for "all" auth methods or a specific auth method (userpass, ldap or approle).
```

File: website/content/docs/configuration/user-lockout.mdx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+User lockouts configuration is done through the Vault configuration file using
+the `user_lockout` stanza:
+
+```hcl
+user_lockout [NAME] {
+  [PARAMETERS...]
+}
+```
+
+For example:
```
Comment (mladlow) [inline website/content/docs/configuration/user-lockout.mdx:35]: Great example!

**Review by mladlow — COMMENTED** [review]
@akshya96 - I left some small edit suggestions and one request to re-order a nav bar.

Overall this is beautiful documentation for the new feature. It's very comprehensive and empathetic!

File: website/content/api-docs/system/auth.mdx [hunk 3]
```diff
@@ -346,6 +346,21 @@ can be achieved without `sudo` via `sys/mounts/auth/[auth-path]/tune`._
 - `plugin_version` `(string: "")` – Specifies the semantic version of the plugin
   to use, e.g. "v1.0.0". Changes will not take effect until the mount is reloaded.
 
+- `user_lockout_config` `(map<string|string>: nil)` – Specifies the user lockout configuration
```
[the remark this answers is not in the document]
Comment (akshya96) [reply website/content/api-docs/system/auth.mdx:349]: I will add this to sample request.

File: changelog/18478.txt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
+website: Add documentation about user lockout concept, configuration using config file and tune auth, api and telemetry.
```
[the remark this answers is not in the document]
Comment (akshya96) [reply changelog/18478.txt]: Thanks :) I forgot that, I will remove this changelog entry.

File: website/content/docs/configuration/user-lockout.mdx (hunk 2 above)
Comment (akshya96) [reply website/content/docs/configuration/user-lockout.mdx:35]: Thanks :)

File: website/content/api-docs/system/auth.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (mladlow) [reply website/content/api-docs/system/auth.mdx:349]: Ah, that's helpful too, but I meant the "5s" or "30m" examples in the explanation of the setting. Good addition though.

**Review by mladlow — APPROVED** [review]

File: website/content/api-docs/system/user-lockout.mdx [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+| `POST` | `/sys/locked-users/:mount_accessor/unlock/:alias-identifier` |
+
+### Parameters
+
+- `mount_accessor` `(string, required)` - Specifies the identifier of the auth mount entry to which the user
+belongs
+- `alias_identifier` `(string, required)` - It is the name of the alias (user). For example, if the alias 
+belongs to userpass backend, the name should be a valid username within userpass auth method. If the alias belongs
+to an approle auth method, the name should be a valid RoleID. If the alias belongs to an ldap auth method, the name
+should be a valid username
```
Comment (raskchanky) [inline website/content/api-docs/system/user-lockout.mdx:213]:
```suggestion
should be a valid username.
```

File: website/content/partials/user-lockout.mdx [hunk 6]
```diff
@@ -0,0 +1,23 @@
+If a user provides bad credentials several times in quick succession, 
+Vault will stop trying to validate their credentials for a while, instead returning immediately 
+with a permission denied error. We call this behavior "user lockout". The time for which 
+an user will be locked out is called “lockout duration”. The user will be able to login after the lockout
```
Comment (raskchanky) [inline website/content/partials/user-lockout.mdx:4]:
```suggestion
a user will be locked out is called “lockout duration”. The user will be able to login after the lockout 
```

**Review by raskchanky — APPROVED** [review]
A few nits, looks good.

File: website/content/docs/commands/auth/tune.mdx [hunk 7]
```diff
@@ -43,6 +43,30 @@ $ vault auth tune -audit-non-hmac-request-keys=value1 -audit-non-hmac-request-ke
 Success! Tuned the auth method at: github/
 ```
 
+User lockout feature is only supported for userpass, ldap and approle auth methods.
```
Comment (mpalmi) [inline website/content/docs/commands/auth/tune.mdx:46]: Super nitpicky oxford comma callout. Feel free to ignore:

```suggestion
User lockout feature is only supported for userpass, ldap, and approle auth methods.
```

File: website/content/docs/commands/auth/tune.mdx [hunk 8]
```diff
@@ -87,3 +111,15 @@ flags](/docs/commands) included on all commands.
 - `-plugin-version` `(string: "")` - Configures the semantic version of the plugin
   to use. The new version will not start running until the mount is
   [reloaded](/docs/commands/plugin/reload).
+
+- `-user-lockout-threshold` `(string: "")` - Specifies the number of failed login attempts 
+  after which the user is locked out. User lockout feature was added in Vault 1.13. 
+
+- `-user-lockout-duration` `(duration: "")` - Specifies the duration for which an user will be locked out.
```
Comment (mpalmi) [inline website/content/docs/commands/auth/tune.mdx:118]:
```suggestion
- `-user-lockout-duration` `(duration: "")` - Specifies the duration for which a user will be locked out.
```

File: website/content/docs/concepts/user-lockout.mdx [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+The precedence for user lockout disable is as follows: 
+
+Disable using environment variable VAULT_DISABLE_USER_LOCKOUT >> 
+Configuration for an auth mount using tune >> Configuration for an auth method in config file >> 
+Configuration for "all" auth methods in config file >> Default values.
+
+## Configuration
+
+User lockout parameters can be configured using config file for "all" auth methods or a specific auth method (userpass, ldap or approle).
```
Comment (mpalmi) [inline website/content/docs/concepts/user-lockout.mdx:29]: nit:

```suggestion
User lockout parameters can be configured using config file for "all" auth methods or a specific auth method (userpass, ldap, or approle).
```

**Review by mpalmi — APPROVED** [review]
Really nice docs. I'll probably use this as a reference next time I need to document a feature! 😄

File: website/content/partials/user-lockout.mdx [hunk 10]
```diff
@@ -0,0 +1,23 @@
+If a user provides bad credentials several times in quick succession, 
+Vault will stop trying to validate their credentials for a while, instead returning immediately 
+with a permission denied error. We call this behavior "user lockout". The time for which 
+an user will be locked out is called “lockout duration”. The user will be able to login after the lockout 
+duration has passed. The number of failed login attempts after which the user is locked out is called 
+“lockout threshold”. The lockout threshold counter is reset to zero after a few minutes without login attempts,
```
[the remark this answers is not in the document]
Comment (akshya96) [reply website/content/partials/user-lockout.mdx:6]: This is just an understanding of what user lockout feature is. This is used in concepts page and we have linked configuration, auth tune and apis there. "lockout threshold" can be tuned or configured I feel like we cannot just link it to one of them.

**Pull request merged**